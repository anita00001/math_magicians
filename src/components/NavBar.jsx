import { Link } from 'react-router-dom';
import '../styles/NavBar.css';

const links = [
  { path: '/', text: 'Home' },
  { path: '/calculator', text: 'Calculator' },
  { path: '/quote', text: 'Quote' },
];

const NavBar = () => (
  <header className="navbar-container">
    <nav className="nav-bar">
      <h1 className="logo"><Link to="/">Math Magicians</Link></h1>
      <ul className="nav-link">
        {links.map((link) => (
          <li key={link.text}>
            <Link to={link.path}>{link.text}</Link>
          </li>
        ))}
      </ul>
    </nav>
  </header>
);

export default NavBar;
