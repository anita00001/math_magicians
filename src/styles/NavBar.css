.navbar-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.nav-bar {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

h1 :-webkit-any-link {
  text-decoration: none;
  color: rgb(132, 23, 23);
}

.nav-link {
  display: flex;
  gap: 1em;
}

.nav-link li {
  list-style: none;
}

.nav-link li:nth-child(2) {
  padding: 0 0.5em;
  border-right: 1px solid black;
  border-left: 1px solid black;
}
