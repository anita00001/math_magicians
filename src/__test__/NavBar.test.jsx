import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import NavBar from '../components/NavBar';

describe('NavBar component', () => {
  test('renders correctly', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    const tree = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    ).toJSON();
    const navbars = screen.getAllByRole('link');
    expect(navbars).toHaveLength(4);
  });
});
