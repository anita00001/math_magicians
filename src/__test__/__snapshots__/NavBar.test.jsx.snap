// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders correctly 1`] = `
<header
  className="navbar-container"
>
  <nav
    className="nav-bar"
  >
    <h1
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Math Magicians
      </a>
    </h1>
    <ul
      className="nav-link"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
